aiida/orm/nodes/data/code/abstract.py

```diff
@@ -76,6 +77,14 @@ def get_executable(self) -> str:
         :return: The executable to be called in the submission script.
         """
 
+    @abc.abstractmethod
+    def presubmit_check(self, folder: Folder):
+        """check the validity when calling in presubmit.
+
+        :param folder: a SandboxFolder that can be used to write calculation input files and the scheduling script.
+        :raises PluginInternalError: The plugin created a file that is also the executable name!.
+        """
```

Comment:
Thinking about this a bit more, I think we should probably just make this method concrete on the base class and just have it `pass`. A subclass can simply override it if needed. Also I wonder if we need to make the name more specific. It currently really serves as a check that the contents of the prepared running directory are compatible with the `Code`, so maybe we should name it more after that. How about the following:
```suggestion
    def validate_working_directory(self, folder: SandboxFolder):
        """Validate content of the working directory created by the :class:`~aiida.engine.CalcJob` plugin.

        This method will be called by :meth:`~aiida.engine.processes.calcjobs.CalcJob.presubmit` when a new calculation
        job is launched, passing the :class:`~aiida.common.folders.SandboxFolder` that was used by the plugin used for
        the calculation to create the input files for the working directory. This method can be overridden by
        implementations of the ``AbstractCode`` class that need to validate the contents of that folder.

        :param folder: A sandbox folder that the ``CalcJob`` plugin wrote input files to that will be copied to the
            working directory for the corresponding calculation job instance.
        :raises PluginInternalError: If the content of the sandbox folder is not valid.
        """
```